src/OrbitGl/MainWindowInterface.h

```diff
@@ -20,7 +23,8 @@ namespace orbit_gl {
 class MainWindowInterface {
  public:
   virtual void ShowTooltip(std::string_view message) = 0;
-  virtual void ShowSourceCode(const std::filesystem::path& file_path, size_t line_number) = 0;
+  virtual void ShowSourceCode(const std::filesystem::path& file_path, size_t line_number,
+                              std::optional<std::unique_ptr<CodeReport>> code_report) = 0;
```

Comment:
`optional`+`unique_ptr` looks particularly ugly to me. Is it a problem to only use `unique_ptr` and just pass a `unique_ptr` holding `nullptr` when the `CodeReport` is not available? If you rename to `maybe_code_report` like in the definition of `OrbitMainWindow::ShowSourceCode` then it should be clear enough.

Reply:
I know, it's ugly. But it's so much better than having bugs because of unexpected nullpointers. And we had a lot of them.

I will change the name!

Reply:
I wanted to suggest `std::optional<CodeReport>`, but since this is not possible, I like `std::optional<std::unique_ptr<CodeReport>>` more than just `unique_ptr`